ext/RMagick/rmimage.c

```diff
@@ -2307,7 +2503,13 @@ Image_channel_entropy(int argc, VALUE *argv, VALUE self)
     }
 
     exception = AcquireExceptionInfo();
+#if IMAGEMAGICK_7
+    BEGIN_CHANNEL_MASK(image, channels);
+    (void) GetImageEntropy(image, &entropy, exception);
+    END_CHANNEL_MASK(image);
```

Comment:
I would not prefer that. At some point we will remove the IM6 code and then we would need to move those changes into rmimage.c I would really prefer to keep it this way.

Reply:
I have also thought a lot about this and this was the most maintainable solution that I could come up with. There is a big difference in the method signatures because most IM7 methods now don't have a channel version of the method and have an exceptioninfo argument. But the setup and checks in RMagick are the same for most of these calls. Most changes have been isolated to how the ImageMagick API is being called.